addons/sms/wizard/sms_composer.py

```diff
@@ -203,7 +203,7 @@ def _action_send_sms_comment_single(self, records=None):
         # on the numbers in the database.
         records = records if records is not None else self._get_records()
         records.ensure_one()
-        if self.recipient_single_number_itf and self.recipient_single_number_itf != self.recipient_single_number:
+        if self.recipient_single_number_itf and self.recipient_single_number_itf != self.recipient_single_number and self.number_field_name:
```

Comment:
Apparently we assume there should be 'numbers' or a number field in ``_action_send_sms_comment`` :)

I think just updating numbers should be fine here.

```suggestion
        if not self.number_field_name:
            self.numbers = self.recipient_single_number_itf or self.recipient_single_number
        elif self.recipient_single_number_itf and self.recipient_single_number_itf != self.recipient_single_number:
```